### PR TITLE
Upgrade to mcli0.4, smaller mcli improvements

### PR DESCRIPTION
**Pull request by aspfohl**
# What does this PR do?

Upgrades to mcli 0.4.0

Small changes:
* Uses root import instead of `mcli.sdk` (This is not required, but feels nicer 🙂)
* Renames mcp to mcli
* Removes `mcli init --mcloud` for CI testing - this is only user facing, not actually needed if api key env variable is set

# What issue(s) does this change relate to?

Dependabot wants to update since we recently pushed 0.4.0 https://github.com/mosaicml/composer/pull/2220

### CODE REVIEW

File: composer/loggers/mosaicml_logger.py [hunk 1]
```diff
@@ -113,8 +113,8 @@ def _log_metadata(self, metadata: Dict[str, Any]) -> None:
     def _flush_metadata(self, force_flush: bool = False) -> None:
         """Flush buffered metadata to MosaicML if enough time has passed since last flush."""
         if self._enabled and (time.time() - self.time_last_logged > self.log_interval or force_flush):
+            from mcli import update_run_metadata
             from mcli.api.exceptions import MAPIException
```
Comment (mvpatel2000) [inline composer/loggers/mosaicml_logger.py:117]: Can we direct import the exception too?

Comment (aspfohl) [reply composer/loggers/mosaicml_logger.py]: Done.

**Comment by mvpatel2000** [comment]
@aspfohl can you please fix unit tests and then we'll merge?

**Comment by mvpatel2000** [comment]
Note: In the logger, we now do `mcli.update_run_metadata` so the unit test can actually monkeypatch it. Otherwise, it gets imported and we can't actually change it without something a lot more complicated (monkeypatching the body of the fn without changing the fn pointer, which is nasty)

**Review by mvpatel2000 — APPROVED** [review]

**Review by dakinggg — APPROVED** [review]
approving to unblock

**Pull request merged**